qa/workunits/rbd/luks-encryption.sh

```diff
@@ -42,16 +43,63 @@ function test_encryption_format() {
   # write via librbd && compare
   sudo dd if=/tmp/testdata1 of=$LIBRBD_DEV conv=fdatasync
   drop_caches
```

Comment:
Perhaps not, but in theory cryptsetup is free to read-ahead to buffer cache when loading.
To be on the safe side I prefer to keep the drop_caches, as saying "assert that the buffer cache is empty".

Reply:
What drop_caches does is it makes one think that the buffer cache is involved somehow.  But it's not, since you are comparing the file which is written to at test setup time and remains immutable for the duration of the test (/tmp/testdata1) and the file which contains data that has just been read directly from disk (/tmp/cmpdata).  iflag=direct should trump anything that might still linger in any cache.

Reply:
OK, given that:
- all writes are done via `conv=fdatasync`
- all reads are done via `iflag=direct`
- cryptsetup device reads/writes are done to /dev/nbdX (which maps to librbd), and it is unknown to me if they require a flush

Should I remove all calls do `drop_caches` (there are currently 3 of those)?
Or should I keep only some? which?

Reply:
I'd keep the first one, because there you are scribbling "through" `RAW_DEV` with `rbd encryption format`  .  The following reads from `RAW_DEV` performed by `cryptsetup open` may need it.  The other two seem redundant to me.

For extra clarity, I'd replace `conv=fdatasync` with `oflag=direct` but it's up to you.  Either should be fine.